lib/actual_db_schema/git_hooks.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require "fileutils"
+
+module ActualDbSchema
+  # Handles the installation of a git post-checkout hook that rolls back phantom migrations when switching branches
+  module GitHooks
+    extend ActualDbSchema::OutputFormatter
+
+    POST_CHECKOUT_HOOK = <<~BASH
+      #!/usr/bin/env bash
+      # ActualDbSchema post-checkout hook
+      # This hook runs whenever you switch branches with `git checkout` to rollback phantom migrations.
+
+      if [ -f ./bin/rails ]; then
+        if [ -n "$ACTUAL_DB_SCHEMA_GIT_HOOKS_ENABLED" ]; then
+          GIT_HOOKS_ENABLED="$ACTUAL_DB_SCHEMA_GIT_HOOKS_ENABLED"
+        else
+          GIT_HOOKS_ENABLED=$(./bin/rails runner "puts ActualDbSchema.config[:git_hooks_enabled]" 2>/dev/null)
+        fi
+
+        if [ "$GIT_HOOKS_ENABLED" == "true" ]; then
+          ./bin/rails db:rollback_branches
+        fi
+      fi
+    BASH
+
+    class << self
+      def install_post_checkout_hook
+        return unless git_hooks_enabled?
+        return unless hooks_directory_present?
+
+        hook_path = hooks_dir.join("post-checkout")
+
+        return unless hook_exist?(hook_path)
```

Comment:
If safe installation is not possible, can we provide instructions to the user on how to install it manually? For instance, we could show the content of the `POST_CHECKOUT_HOOK` along with some guidance on how to use it.

Another idea: what if we include specific markers in the post-checkout file, such as comment lines that encapsulate our script? This way, we could insert only the necessary lines of code. Later, the installation process could even replace the existing lines. This approach would allow us to apply patches to the bash code in the future. Before inserting the lines, we should scan the hook file. If the user has already added these lines manually, we should avoid inserting them automatically.